src/v/cluster/partition.cc

```diff
@@ -542,6 +541,10 @@ ss::future<> partition::start(std::optional<topic_configuration> topic_cfg) {
         }
     }
 
+    // Start the probe after the partition is fully initialised, but before
+    // starting everything.
+    _probe.setup_metrics(ntp);
+
```

Comment:
If this is required for correctness of the tests, should this commit be moved earlier?

Reply:
done